pkg/networkservice/core/next/context.go

```diff
@@ -48,6 +48,9 @@ func withNextServer(parent context.Context, next networkservice.NetworkServiceSe
 // Server -
 //   Returns the Server networkservice.NetworkServiceServer to be called in the chain from the context.Context
 func Server(ctx context.Context) networkservice.NetworkServiceServer {
+	if ctx == nil {
```

Comment:
Question... why return nil here instead of tailServer?

Reply:
actually it can be nil in two cases:
1. At the first call of method request/close with `ctx = nil`. In this case, we looking for the next parent chain and value can be nil
2. Call this method outside of the chain. I.e. user not used `next.NewServer()`

Reply:
Finally, for the second case I think it will be better just throw nil panic exception. Fixed by https://github.com/networkservicemesh/sdk/pull/173/commits/1d5588cc84d32b48e825132a8f5912d8c3102afd

Reply:
Why would we panic if called outside of a chain?

Reply:
1. We have analogies here:
https://github.com/networkservicemesh/sdk/blob/master/pkg/networkservice/core/trace/context.go#L43
https://github.com/networkservicemesh/sdk-vppagent/blob/master/pkg/networkservice/vppagent/context.go#L50
...
2. I think when the user tries to use nil context to get the value it is an invalid case. And we should not work with invalid states.

Reply:
@edwarnicke Is it make sense?

Reply:
That's fair :)